addons/spreadsheet_dashboard/models/spreadsheet_dashboard.py

```diff
@@ -39,3 +40,12 @@ def _empty_workbook_data(self):
     def _compute_raw(self):
         for dashboard in self:
             dashboard.raw = base64.decodebytes(dashboard.data)
+
+    @api.onchange('data')
+    def _onchange_data_(self):
```

Comment:
> what about an `@api.constrains` instead ? Also: you can/should add a test

Indeed it would work. But, according to me, the user should get the error immediately after he uploads the wrong file. But if you want we can go with the `@api.constrains`. What do you say?

Reply:
You've got a point. Let's keep it that way. (And I have other ideas for a better validation in master anyway :smile: )